### PR TITLE
Add and update functionality to make SDK consistent with other server SDKs

### PR DESCRIPTION
**Pull request by hamchapman**
### Why?

----

- [x] CHANGELOG updated if relevant?

### CODE REVIEW

**Comment by shrugs** [comment]
There are some functions that returns JSON-parsed responses (like `get_roles`) and others that return raw requests (like `get_users`). It would be good to standardize on one pattern for requests that return information.

**Comment by hamchapman** [comment]
I was thinking about this yesterday and I think the standard we should settle on is returning a hash like this:

```ruby
{
  status: 200,
  headers: {
    ...
  },
  body: {
    roles: {
      ...
    }
  }
```

so the body is a JSON-parsed body if there was a response body. If not then the body will be nil.

If the status isn't in the 200-299 range then you get back an `PusherPlatform::ErrorResponse` object: https://github.com/pusher/pusher-platform-ruby/blob/master/lib/pusher-platform/error_response.rb#L5

WDYT?

**Comment by hamchapman** [comment]
In fact currently the `PusherPlatform::ErrorResponse` object is raised as an error. Is that behaviour desirable, i.e. should it still raise? Or is just returning the error body more useful? I feel like the current behaviour is more useful so I suggest we keep it raising

File: examples/create_room.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative '../lib/chatkit'
+
+chatkit = Chatkit::Client.new({
+  instance_locator: "v1:us1:some-instance-id",
+  key: "the-id-bit:the-secret-bit"
+})
+
+# p chatkit.create_room({
```
[the remark this answers is not in the document]
Comment (hamchapman) [reply examples/create_room.rb]: Leaving it as a demo for how you can provide more info

**Comment by hamchapman** [comment]
@shrugs there's now a `Chatkit::Error` type that all Chatkit-related (and underlying PusherPlatform errors get re-raised as) errors inherit from so you can rescue just using that, as you have in your example code 👍

**Comment by hamchapman** [comment]
Replaced by #16

**Pull request closed without merging**